File: leetcode/LinkedListTests/LinkedList_19.cs
namespace LinkedListTests;

[TestFixture]
class LinkedList_19
{
    class Solution {
        public ListNode RemoveNthFromEnd(ListNode head, int n) {
            var fast = head;
            for(var i=0; i<n; i++) {
                fast = fast.next;
            }
            var dummyHead = new ListNode(0);
            var slow =dummyHead;
            slow.next = head;
            while(fast is not null) {
                slow = slow.next;
                fast = fast.next;
            }
            slow.next = slow.next.next;
            return dummyHead.next;
        }
    }
    
}

File: leetcodePython/LinkedList/linkedlist_19.py
from typing import Optional


class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        fast = head
        for _ in range(n):
            fast = fast.next
        dummy = ListNode(-1)
        dummy.next = head
        slow = dummy
        while fast:
            slow = slow.next
            fast = fast.next
        slow.next = slow.next.next
        return dummy.next
